vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/CheckboxSteps.java

```diff
@@ -18,274 +18,113 @@
 import org.jbehave.core.annotations.When;
 import org.openqa.selenium.WebElement;
-import org.openqa.selenium.WrapsElement;
 import org.vividus.bdd.monitor.TakeScreenshotOnFailure;
 import org.vividus.bdd.steps.ui.web.validation.IBaseValidations;
 import org.vividus.selenium.element.Checkbox;
 import org.vividus.ui.web.State;
 import org.vividus.ui.web.action.CheckboxAction;
 import org.vividus.ui.web.action.IMouseActions;
-import org.vividus.ui.web.action.search.ActionAttributeType;
 import org.vividus.ui.web.action.search.SearchAttributes;
-import org.vividus.ui.web.action.search.SearchParameters;
-import org.vividus.ui.web.util.LocatorUtil;
 
 @TakeScreenshotOnFailure
-public class CheckboxSteps
+public final class CheckboxSteps
 {
     private static final String THE_FOUND_CHECKBOX_IS = "The found checkbox is ";
-    private static final String CHECKBOX_WITH_NAME = "Checkbox with name '%s'";
     private static final String CHECKBOX = "Checkbox";
-    private static final String CHECKBOX_LOCATOR = LocatorUtil.getXPath("input[@type='checkbox']");
-    private static final String CHECKBOX_WITH_ATTRIBUTE = "Checkbox with the attribute '%1$s'='%2$s'";
 
-    @Inject private IBaseValidations baseValidations;
-    @Inject private IMouseActions mouseActions;
+    private final IBaseValidations baseValidations;
+    private final IMouseActions mouseActions;
 
-    /**
-     * Checks checkbox within the context
-     * <p>Actions performed at this step:</p>
-     * <ul>
-     * <li>Finds a checkbox within the context;</li>
-     * <li>Checks its state, if it's <b><i>not selected</i></b> then changes its state
-     * <li>Waits for page to load;</li>
-     * </ul>
-     */
-    @When("I check a checkbox")
-    public void checkCheckBox()
+    private CheckboxSteps(IBaseValidations baseValidations, IMouseActions mouseActions)
     {
-        WebElement checkBox = baseValidations.assertIfElementExists(CHECKBOX, new SearchAttributes(
-                ActionAttributeType.XPATH, CHECKBOX_LOCATOR));
-        changeCheckboxState(new Checkbox(checkBox), true);
+        this.baseValidations = baseValidations;
+        this.mouseActions = mouseActions;
     }
 
     /**
-     * Checks all enabled checkboxes within the set context
-     * <p>Actions performed at this step:</p>
-     * <ul>
-     * <li> Verifies that at least 1 checkbox is found within the search context;</li>
-     * <li> Checks the state of each checkbox found on the page, if it's <b><i>not selected</i></b>
-     * then change it to <b><i>selected</i></b>;</li>
-     * <li>Waits for page to load;</li>
-     * </ul>
+     * Performs action on all the checkboxes found by locator
+     * @param checkBoxAction Actions to be performed (CHECK, UNCHECK)
+     * @param checkboxesLocator Locator to locate checkboxes
      */
-    @When("I check all the checkboxes")
-    public void checkAllCheckboxes()
+    @When("I $checkboxState all checkboxes located by `$checkboxesLocator`")
```

Comment:
what about converting the step to composite one?

Reply:
Could not be achieved without research overhead per checkbox